File: post-service/src/main/java/com/example/domain/Comment.java
package com.example.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "comments")
public class Comment implements Serializable {

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "post_id")
    private Post post;
    private String content;
    private LocalDateTime createdAt;
}

File: r2dbc-kotlin-co/src/main/kotlin/com/example/Comment.kt
package com.example

import com.fasterxml.jackson.annotation.JsonIgnore
import io.micronaut.data.annotation.*
import io.micronaut.data.model.naming.NamingStrategies
import java.time.LocalDateTime
import java.util.*

@MappedEntity(value = "comments", namingStrategy = NamingStrategies.UnderScoreSeparatedLowerCase::class)
data class Comment(
    @AutoPopulated//generated value UUID does not work here.
    @field:Id var id: UUID? = null,
    var content: String,
    @field:DateCreated var createdAt: LocalDateTime? = LocalDateTime.now(),
    @field:Relation(Relation.Kind.MANY_TO_ONE) @field:MappedProperty("post_id") @field:JsonIgnore var post: Post? = null
)